gammapy/utils/roots.py

```diff
@@ -6,7 +6,12 @@
 import astropy.units as u
 from gammapy.utils.interpolation import interpolation_scale
 
-BAD_RES = RootResults(root=np.nan, iterations=0, function_calls=0, flag=0)
+try:
```

Comment:
Why is the try needed? Why does this yield a `TypeError` in some cases?

Reply:
method is a new argument in the dev version, is there a better way to do this ?